devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusPrepare.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.gradle.tasks;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.plugins.Convention;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskAction;
+
+import io.quarkus.bootstrap.BootstrapException;
+import io.quarkus.bootstrap.app.CuratedApplication;
+import io.quarkus.bootstrap.app.QuarkusBootstrap;
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.bootstrap.model.AppArtifact;
+import io.quarkus.bootstrap.resolver.AppModelResolver;
+import io.quarkus.deployment.CodeGenerator;
+
+public class QuarkusPrepare extends QuarkusTask {
+
+    public static final String INIT_AND_RUN = "initAndRun";
+    private Path sourcesDirectory;
+    private Consumer<Path> sourceRegistrar;
+    private boolean test = false;
+
+    public QuarkusPrepare() {
+        super("Quarkus performs pre-build preparations, such as sources generation");
+    }
+
+    @TaskAction
+    public void prepareQuarkus() {
+        getLogger().lifecycle("preparing quarkus application");
+
+        final AppArtifact appArtifact = extension().getAppArtifact();
+        appArtifact.setPaths(QuarkusGradleUtils.getOutputPaths(getProject()));
```

Comment:
Was this necessary for dev mode? This is something to be avoided.

Reply:
I ran the tests I have and they work without it. I'm dropping it.